libethereum/BlockChain.cpp

```diff
@@ -194,71 +194,78 @@ void BlockChain::init(ChainParams const& _p)
+    auto const chainSubPathBlocks = chainPath / fs::path("blocks");
+    auto const extrasPath = chainPath / fs::path(toString(c_databaseVersion));
+    auto const extrasSubPathExtras = extrasPath / fs::path("extras");
     unsigned lastMinor = c_minorProtocolVersion;
 
     if (db::isDiskDatabase())
     {
         fs::create_directories(extrasPath);
         DEV_IGNORE_EXCEPTIONS(fs::permissions(extrasPath, fs::owner_all));
 
-        bytes status = contents(extrasPath / fs::path("minor"));
+        auto const extrasSubPathMinor = extrasPath / fs::path("minor");
+        bytes const status = contents(extrasSubPathMinor);
         if (!status.empty())
             DEV_IGNORE_EXCEPTIONS(lastMinor = (unsigned)RLP(status));
         if (c_minorProtocolVersion != lastMinor)
         {
-            cnote << "Killing extras database (DB minor version:" << lastMinor << " != our miner version: " << c_minorProtocolVersion << ").";
+            cnote << "Killing extras database " << extrasPath << " (DB minor version:" << lastMinor
+                  << " != our minor version: " << c_minorProtocolVersion << ").";
             DEV_IGNORE_EXCEPTIONS(fs::remove_all(extrasPath / fs::path("details.old")));
-            fs::rename(extrasPath / fs::path("extras"), extrasPath / fs::path("extras.old"));
+            fs::rename(extrasSubPathExtras, extrasPath / fs::path("extras.old"));
             fs::remove_all(extrasPath / fs::path("state"));
-            writeFile(extrasPath / fs::path("minor"), rlp(c_minorProtocolVersion));
+            writeFile(extrasSubPathMinor, rlp(c_minorProtocolVersion));
             lastMinor = (unsigned)RLP(status);
         }
 
         if (_we == WithExisting::Kill)
         {
-            cnote << "Killing blockchain & extras database (WithExisting::Kill).";
-            fs::remove_all(chainPath / fs::path("blocks"));
-            fs::remove_all(extrasPath / fs::path("extras"));
+            cnote << "Killing blockchain (" << chainSubPathBlocks << ") & extras ("
+                  << extrasSubPathExtras << ") databases (WithExisting::Kill).";
+            fs::remove_all(chainSubPathBlocks);
+            fs::remove_all(extrasSubPathExtras);
         }
     }
 
     try
     {
-        m_blocksDB = db::DBFactory::create(chainPath / fs::path("blocks"));
-        m_extrasDB = db::DBFactory::create(extrasPath / fs::path("extras"));
+        m_blocksDB = db::DBFactory::create(chainSubPathBlocks);
+        m_extrasDB = db::DBFactory::create(extrasSubPathExtras);
     }
     catch (db::DatabaseError const& ex)
     {
-        // Check the exact reason of errror, in case of IOError we can display user-friendly message
-        if (*boost::get_error_info<db::errinfo_dbStatusCode>(ex) != db::DatabaseStatus::IOError)
-            throw;
+        // Determine which database open call failed
+        auto const dbPath = !m_blocksDB.get() ? chainSubPathBlocks : extrasSubPathExtras;
+        cerr << "Error opening database: " << dbPath;
```

Comment:
This would just output to stderr, we should use `cerror` for proper boost.log logging here
```suggestion
        cerror << "Error opening database: " << dbPath;
```
in other places below, too

Reply:
Ah, thanks for calling this out!